### DISCUSSION TITLE
Belongs to Many Relation Manager EditForm Method

### DISCUSSION THREAD

**Discussion by intrepidws** [filamentphp/filament#2193] [Ideas]
A Belongs to Many Relation Manager allows you to define a form to be used specifically for attaching a record, using the `attachForm()` method.

I'm wondering what people would think of a method that would allow you to define a form to be used specifically for editing a record, something like an `editForm()` method.

In my use case, I want people to be able to create a model using one form, attach a model using another form and edit a model using another form. In the case of the edit, I really only want them to be able to edit the pivot table fields whereas I want them to have all of the fields available (both on the relationship model and the pivot table).

Would this be a useful addition?

**Comment by danharrin** [comment]
Yeah I think it would be a useful addition, although you could also use `->hidden(fn (RelationManager $livewire) => $livewire->mountedTableAction === 'edit')` to hide fields / component groups on the edit form

**Comment by intrepidws** [comment]
I tried my hand at implementing an `editForm()` method but I fear it's a bit beyond my capabilities. I'll try to give it another shot next week, since I kinda hate having to resort to sprinkling the `hidden()` method throughout. It feels much cleaner to have separate methods.

**Comment by danharrin** [comment]
As of v2.13, you can now pass a custom form to each Action. `->form([...])`

**Reply by intrepidws** [reply]
That's super helpful, thanks Dan!
